dateparser/conf.py

```diff
@@ -1,67 +1,79 @@
 # -*- coding: utf-8 -*-
+import hashlib
 from pkgutil import get_data
+from itertools import chain
+from functools import wraps
+import six
 
 from yaml import load as load_yaml
 
-"""
-:mod:`dateparser`'s parsing behavior can be configured like below
+from .utils import registry
 
-*``PREFER_DAY_OF_MONTH``* defaults to ``current`` and can have ``first`` and ``last`` as values::
 
-    >>> from dateparser.conf import settings
-    >>> from dateparser import parse
-    >>> parse(u'December 2015')
-    datetime.datetime(2015, 12, 16, 0, 0)
-    >>> settings.update('PREFER_DAY_OF_MONTH', 'last')
-    >>> parse(u'December 2015')
-    datetime.datetime(2015, 12, 31, 0, 0)
-    >>> settings.update('PREFER_DAY_OF_MONTH', 'first')
-    >>> parse(u'December 2015')
-    datetime.datetime(2015, 12, 1, 0, 0)
+@registry
+class Settings(object):
+    """Control and configure default parsing behavior of dateparser.
+    Currently, supported settings are:
+    - `PREFER_DATES_FROM`: defaults to `current_period`. Options are `future` or `past`.
+    - `SUPPORT_BEFORE_COMMON_ERA`: defaults to `False`.
+    - `PREFER_DAY_OF_MONTH`: defaults to `current`. Could be `first` and `last` day of month.
+    - `SKIP_TOKENS`: defaults to `['t']`. Can be any string.
+    """
 
-*``PREFER_DATES_FROM``* defaults to ``current_period`` and can have ``past`` and ``future`` as values.
-Assuming current date is June 16, 2015::
+    _default = True
+    _yaml_data = {}
 
-    >>> from dateparser.conf import settings
-    >>> from dateparser import parse
-    >>> parse(u'March')
-    datetime.datetime(2015, 3, 16, 0, 0)
-    >>> settings.update('PREFER_DATES_FROM', 'future')
-    >>> parse(u'March')
-    datetime.datetime(2016, 3, 16, 0, 0)
+    def __init__(self, **kwargs):
+        self._updateall(
+            chain(self._get_settings_from_yaml().items(),
+                  kwargs.items())
+        )
 
-*``SKIP_TOKENS``* is a ``list`` of tokens to discard while detecting language. Defaults to ``['t']`` which skips T in iso format datetime string.e.g. ``2015-05-02T10:20:19+0000``.
-This only works with :mod:`DateDataParser` like below:
+    @classmethod
+    def get_key(cls, *args, **kwargs):
+        if not args and not kwargs:
+            return 'default'
 
-    >>> settings.update('SKIP_TOKENS', ['de'])  # Turkish word for 'at'
-    >>> from dateparser.date import DateDataParser
-    >>> DateDataParser().get_date_data(u'27 Haziran 1981 de')  # Turkish (at 27 June 1981)
-    {'date_obj': datetime.datetime(1981, 6, 27, 0, 0), 'period': 'day'}
-"""
+        keys = sorted(['%s-%s' % (key, str(kwargs[key])) for key in kwargs])
+        return hashlib.md5(''.join(keys)).hexdigest()
 
+    @classmethod
+    def _get_settings_from_yaml(cls):
+        if not cls._yaml_data:
+            data = get_data('data', 'settings.yaml')
+            cls._yaml_data = load_yaml(data).pop('settings', {})
```

Comment:
Just a **nitpick**: as initial `_yaml_data` equals `{}`, I'd expect `update` here. Or, another way to look at it, as we have assignment here, I'd expect `_yaml_data` be not initialized (`None`) :)